packages/features/auth/lib/next-auth-options.ts

```diff
@@ -735,16 +766,25 @@ export const AUTH_OPTIONS: AuthOptions = {
           return "/auth/error?error=use-identity-login";
         }
 
+        // Associate with organization if enabled by flag
+        const [orgUsername, orgId] = await checkIfUserShouldBelongToOrg(user.email);
+
         const newUser = await prisma.user.create({
           data: {
             // Slugify the incoming name and append a few random characters to
             // prevent conflicts for users with the same name.
-            username: usernameSlug(user.name),
+            username: orgId ? orgUsername : usernameSlug(user.name),
```

Comment:
We use forced org username calculated in function if there is an orgId which in turn means autolink is turned on